imap_processing/utils.py

```diff
@@ -0,0 +1,94 @@
+import collections
+
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+
+def sort_by_time(packets, time_key):
+    """Sort packets by specified key.
+
+    Parameters
+    ----------
+    packets : list
+        Decom data packets
+    time_key : str
+        Key to sort by. Must be a key in the packets data dictionary.
+        e.g. "SHCOARSE" or "MET_TIME" or "ACQ_START_COARSE"
+
+    Returns
+    -------
+    list
+        sorted packets
+    """
+    sorted_packets = sorted(packets, key=lambda x: x.data[time_key].raw_value)
+    return sorted_packets
+
+
+def group_by_apid(packets: list):
+    """Group data by apid.
+
+    Parameters
+    ----------
+    packets : list
+        packet list
+
+    Returns
+    -------
+    dict
+        grouped data by apid
+    """
+    grouped_packets = collections.defaultdict(list)
+    for packet in packets:
+        apid = packet.header["PKT_APID"].raw_value
+        grouped_packets.setdefault(apid, []).append(packet)
+    return grouped_packets
+
+
+def convert_raw_to_eu(dataset: xr.Dataset, convertion_table_path, packet_name):
```

Comment:
```suggestion
def convert_raw_to_eu(dataset: xr.Dataset, conversion_table_path, packet_name):
```
minor typo

Reply:
I noticed that and changed on the other PR. Will change it here too